lib/palindrome_check.rb

```diff
@@ -1,5 +1,55 @@
 # A method to check if the input string is a palindrome.
 # Return true if the string is a palindrome. Return false otherwise.
 def palindrome_check(my_phrase)
-  raise NotImplementedError
+
+  if my_phrase == nil
+    return false
+  elsif my_phrase.length == 0
+    return true
+  elsif my_phrase.length == 1
+    return true
+  else
+    x = find_start_index(my_phrase, 0)
+    y = find_ending_index(my_phrase, my_phrase.length - 1)
+    start = my_phrase[x]
+    ending = my_phrase[y]
+
+    until x > y
+      if start == ending
+        x = find_start_index(my_phrase, x + 1)
+        start = my_phrase[x]
+        y = find_ending_index(my_phrase, y - 1)
+        ending = my_phrase[y]
+      else
+        return false
+      end
+    end
+    return true
+  end
+
+  # raise NotImplementedError
 end
+
+def find_start_index(my_phrase, index_val)
```

Comment:
As due diligence, consider passing my_phrase.length as the third parameter to this method. Then, in the loop on line 37, you can first confirm that new_index is not out of bounds before checking the character at the index.

Because you're coding in Ruby, the algorithm works without it since at index length, the value will be `nil` and the loop will terminate. However, if you were coding in any other language, your algorithm will need to be more robust and account for index bounds.